app/utils/injectSaga.js

```diff
@@ -9,35 +9,35 @@ import getInjectors from './sagaInjectors';
  *
  * @param {string} key A key of the saga
  * @param {function} saga A root saga that will be injected
- * @param {string} [mode] By default (constants.DAEMON) the saga will be started
- * on component mount and never canceled or started again. Another two options:
- *   - constants.RESTART_ON_REMOUNT — the saga will be started on component mount and
- *   cancelled with `task.cancel()` on component unmount for improved performance,
- *   - constants.ONCE_TILL_UNMOUNT — behaves like 'RESTART_ON_REMOUNT' but never runs it again.
+ * @param {string} [mode] By default (constants.RESTART_ON_REMOUNT) the saga will be started on component mount and
+ * cancelled with `task.cancel()` on component un-mount for improved performance. Another two options:
+ *   - constants.DAEMON—starts the saga on component mount and never cancels it or starts again,
+ *   - constants.ONCE_TILL_UNMOUNT—behaves like 'RESTART_ON_REMOUNT' but never runs it again.
  *
  */
 export default ({ key, saga, mode }) => WrappedComponent => {
   class InjectSaga extends React.Component {
     static WrappedComponent = WrappedComponent;
 
-    static contextTypes = {
-      store: PropTypes.object.isRequired,
-    };
+    static contextType = ReactReduxContext;
 
     static displayName = `withSaga(${WrappedComponent.displayName ||
       WrappedComponent.name ||
       'Component'})`;
 
     constructor(props, context) {
       super(props, context);
+      const { store } = context;
+      this.injectors = getInjectors(store);
+      const { injectSaga } = this.injectors;
 
-      this.injectors = getInjectors(context.store);
-
-      this.injectors.injectSaga(key, { saga, mode }, this.props);
+      injectSaga(key, { saga, mode }, this.props);
     }
 
     componentWillUnmount() {
-      this.injectors.ejectSaga(key);
+      const { ejectSaga } = this.injectors;
+
+      ejectSaga(key);
```

Comment:
Is this change necessary? The behavior is exactly the same.

Reply:
(Same comment for the `constructor` and the equivalent places in `injectReducer.js`.)

Reply:
You are right! Not needed.